eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/primitive/ObjectBooleanHashMap.java

```diff
@@ -1109,6 +1109,14 @@ private int maxOccupiedWithData()
         return Math.min(capacity - 1, capacity / OCCUPIED_DATA_RATIO);
     }
 
+    /**
+     * Rehashes every element in the set into a new backing table of the smallest possible size and eliminating removed sentinels.
+     */
+    public void compact()
+    {
+        this.rehash(this.smallestPowerOfTwoGreaterThan(this.size()));
```

Comment:
There ought to be a check that the new capacity is different than the old capacity. Otherwise, we spend O(n) time and space for no reason.

Reply:
I'm just realizing that your changes are consistent with the code in primitive-primitive maps. I guess we can land this but we ought to fix these problems later.